### PR TITLE
feat(combobox): implemented virtual scroll for combobox

### PR DESCRIPTION
**Pull request by parsoman**
## Description

Provided a virtual scroll for combobox for all type of use cases except group label type
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/combobox/ComboBox.ts [hunk 1]
```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
...
+                        display: this.expanded
+                          ? this.options.length &&
+                            this.filteredOptions.length === 0 &&
+                            this.inputValue &&
+                            this.allowCustomValue
+                            ? "none"
+                            : "block"
+                          : "none",
+                        "z-index": "1"
+                      })}
```
Comment (vinitkumargoel) [inline web-components/src/components/combobox/ComboBox.ts:1463]: Can we simplify the nested ternary statement?

File: web-components/src/components/combobox/ComboBox.ts [hunk 2]
```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
...
+                                      ).map(({ text, matching }) =>
+                                        matching
+                                          ? html`
+                                              <span class="highlight-text">${text}</span>
+                                            `
+                                          : html`
+                                              <span class="selected-label-text">${text}</span>
+                                            `
+                                      )}
+                                </span>
```
Comment (vinitkumargoel) [inline web-components/src/components/combobox/ComboBox.ts:1528]: Can we simplify the nested ternary statement?

**Review by vinitkumargoel — CHANGES_REQUESTED** [review]

**Review by vikcool2pr — APPROVED** [review]

File: web-components/src/components/combobox/ComboBox.ts [hunk 3]
```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
...
+                              >
+                                ${this.isMulti
+                                  ? html`
+                                      <span class="select-option">
+                                        <md-icon name="icon-check_14"></md-icon>
+                                      </span>
+                                    `
+                                  : nothing}
+                                <span part="label" class="select-label">
+                                  ${this.isCustomContent
```
Comment (merajcal) [inline web-components/src/components/combobox/ComboBox.ts:1514]: extract separate component to simplify it

File: web-components/src/components/combobox/ComboBox.ts [hunk 4]
```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
...
-                  )}
+                  ${this.isMulti && this.allowSelectAll && this.expanded ? this.getSelectAllOption() : nothing}
+                  ${!this.checkForVirtualScroll()
+                    ? repeat(
+                        this.filterOptions(this.trimSpace ? this.inputValue.replace(/\s+/g, "") : this.inputValue),
+                        (option: string | OptionMember) => this.getOptionId(option),
+                        (option: string | OptionMember, optionIndex) => {
+                          if (typeof option !== "string" && this.isOptGroup && option.isLabel === "true") {
+                            return html`
+                              <div
```
Comment (merajcal) [inline web-components/src/components/combobox/ComboBox.ts:1473]: is it possible to extract seprate component ofr each if/else condition for better readability

**Review by merajcal — CHANGES_REQUESTED** [review]

File: web-components/src/components/combobox/ComboBox.ts (hunk 4 above)
Comment (parsoman) [reply web-components/src/components/combobox/ComboBox.ts]: Fixed

File: web-components/src/components/combobox/ComboBox.ts (hunk 3 above)
Comment (parsoman) [reply web-components/src/components/combobox/ComboBox.ts]: Fixed

File: web-components/src/components/combobox/ComboBox.ts (hunk 2 above)
Comment (parsoman) [reply web-components/src/components/combobox/ComboBox.ts]: Fixed

File: web-components/src/components/combobox/ComboBox.ts (hunk 1 above)
Comment (parsoman) [reply web-components/src/components/combobox/ComboBox.ts]: Fixed

**Review by merajcal — APPROVED** [review]

**Review by vinitkumargoel — DISMISSED** [review]

**Pull request merged**